<a href="https://colab.research.google.com/github/NRG367GSI/GitInstructionWorkshop_2022_08_01/blob/main/Homework_11_Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
# Библиотека для работы с табличными данными
import pandas as pd

In [3]:
# Прочтем файл .csv с помощью библиотеки pandas
df = pd.read_csv('/content/drive/MyDrive/titanic/train.csv') # загрузим данные Титаника в объект Pandas DataFrame
tsd = pd.read_csv('/content/drive/MyDrive/titanic/test.csv') # загрузим тестовые данные 

In [4]:
# Посмотреть первые 5 строк 
df.head(5) # train

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
tsd.head(5) # test

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

In [6]:
td = pd.concat([df, tsd], ignore_index=True, sort = False) # объеденим данные из набора train и test

In [7]:
td.head(5)

PassengerId  Survived  Pclass  \
0            1       0.0       3   
1            2       1.0       1   
2            3       1.0       3   
3            4       1.0       1   
4            5       0.0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [8]:
df.shape, tsd.shape, td.shape # В тестовом наборе на 1 колонку меньше отсутствует Survived т.е. выжившие, потому что в тестовом наборе только выжившие.

((891, 12), (418, 11), (1309, 12))

In [9]:
df.isnull().sum() # Колличество нулевых значений в столбцах

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

* PassengerId - идентификатор пассажира
* Survived - Выжил
* Pclass - Класс
* Name - Имя
* Sex - пол
* Age - возрост
* SibSp - число родственников на борту
* Parch - Количество родителей/детей на борту
* Ticket - Номер билета
* Fare - Пассажирский тариф
* Cabin - Кабина
* Embarked - Посадка - порт посадки (C = Шербур; Q = Квинстаун; S = Саутгемптон)

In [10]:
df.columns # список названий колонок таблици

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [11]:
df.describe() # Выводим столбци с числовыми значениями

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [12]:
all_pasenger = df['Survived'].shape[0] # Всего пассажиров
all_pasenger

891

In [13]:
all_surv = df[df['Survived'] != False].shape[0] # Всего выжевших
all_surv

342

In [14]:
all_surv/all_pasenger * 100 # % выжевших

38.38383838383838

In [15]:
pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1)].shape[0] # Колличество выживших первого класса
pclass_1_surv

136

In [16]:
pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2)].shape[0] # Колличество выживших 2 класса
pclass_2_surv

87

In [17]:
pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3)].shape[0] # Колличество выживших 3 класса
pclass_3_surv

119

Козалось бы пассажиры чем выши класс тем больше выживших должно быть, но нет разница выживших между пассажирами первого класса и третьего не большая.
Нужно проверить связь между следующими параметрами:
* Sex - пол, обычно стараются спости женщин и детей в первую очередь
* Age - возрост, до 12, 12-20, 20-30, 30-40, 50 и выше
* SibSp - число родственников на борту, чем больше родственников тем сложнее спастись, всем вмести собраться, маральные дилемы
* Parch - Количество родителей/детей на борту, мужчины стараются защетить женщин и детей, матеря стораются защетить детей

# Разделим выживших разных классов на мужчин женщин

In [18]:
male_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male")].shape[0] # Колличество выживших первого класса мужчин
female_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female")].shape[0] # Колличество выживших первого класса женщин
male_pclass_1_surv, female_pclass_1_surv

(45, 91)

In [19]:
male_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male")].shape[0] # Колличество выживших 2 класса мужчин
female_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female")].shape[0] # Колличество выживших 2 класса женщин
male_pclass_1_surv, female_pclass_1_surv

(17, 70)

In [20]:
male_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male")].shape[0] # Колличество выживших 3 класса мужчин
female_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female")].shape[0] # Колличество выживших 3 класса женщин
male_pclass_1_surv, female_pclass_1_surv

(47, 72)

# Мужчины из первого класса по возростным группам

In [21]:
male_all_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male")].shape[0] # Колличество мужчие из первого класса
male_all_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male")].shape[0]

male_12_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
male_12_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

male_12_20_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
male_12_20_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

male_20_30_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
male_20_30_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

male_30_40_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
male_30_40_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

male_40_50_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
male_40_50_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

male_50__pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
male_50__pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", male_all_pclass_1, male_all_pclass_1_surv, male_all_pclass_1_surv/male_all_pclass_1 * 100, "%", '\n',
      "до 12 лет", male_12_pclass_1, male_12_pclass_1_surv, male_12_pclass_1_surv/male_12_pclass_1 * 100, "%",'\n',
      "12 - 20 лет", male_12_20_pclass_1, male_12_20_pclass_1_surv, male_12_20_pclass_1_surv/male_12_20_pclass_1 * 100, "%", '\n',
      "20 - 30 лет", male_20_30_pclass_1, male_20_30_pclass_1_surv, male_20_30_pclass_1_surv/male_20_30_pclass_1 * 100, "%", '\n',
      "30 - 40 лет", male_30_40_pclass_1, male_30_40_pclass_1_surv, male_30_40_pclass_1_surv/male_30_40_pclass_1 * 100, "%", '\n',
      "40 - 50 лет", male_40_50_pclass_1, male_40_50_pclass_1_surv, male_40_50_pclass_1_surv/male_40_50_pclass_1 * 100, "%", '\n',
      "50 лет и старше", male_50__pclass_1, male_50__pclass_1_surv, male_50__pclass_1_surv/male_50__pclass_1 *100, "%")

Всего 122 45 36.885245901639344 % 
 до 12 лет 3 3 100.0 % 
 12 - 20 лет 4 1 25.0 % 
 20 - 30 лет 19 9 47.368421052631575 % 
 30 - 40 лет 25 13 52.0 % 
 40 - 50 лет 24 9 37.5 % 
 50 лет и старше 29 6 20.689655172413794 %


# # Мужчины из 2 класса по возростным группам

In [22]:
male_all_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male")].shape[0] # Колличество мужчие из первого класса
male_all_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male")].shape[0]

male__12_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
male__12_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

male_12_20_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
male_12_20_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

male_20_30_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
male_20_30_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

male_30_40_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
male_30_40_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

male_40_50_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
male_40_50_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

male_50__pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
male_50__pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", male_all_pclass_2, male_all_pclass_2_surv, male_all_pclass_2_surv/male_all_pclass_2 * 100, "%", '\n',
      
      "до 12 лет", male__12_pclass_2, male__12_pclass_2_surv, male__12_pclass_2_surv/male__12_pclass_2 * 100, "%", '\n',

      "20 - 30 лет", male_20_30_pclass_2, male_20_30_pclass_2_surv, male_20_30_pclass_2_surv/male_20_30_pclass_2 * 100, "%", '\n',

      "30 - 40 лет", male_30_40_pclass_2, male_30_40_pclass_2_surv, male_30_40_pclass_2_surv/male_30_40_pclass_2 * 100, "%", '\n',

      "40 - 50 лет", male_40_50_pclass_2, male_40_50_pclass_2_surv, male_40_50_pclass_2_surv/male_40_50_pclass_2 * 100, "%", '\n',
      
      "50 лет и старше", male_50__pclass_2, male_50__pclass_2_surv, male_50__pclass_2_surv/male_50__pclass_2 *100, "%")

Всего 108 17 15.74074074074074 % 
 до 12 лет 9 9 100.0 % 
 20 - 30 лет 33 0 0.0 % 
 30 - 40 лет 26 3 11.538461538461538 % 
 40 - 50 лет 9 1 11.11111111111111 % 
 50 лет и старше 13 1 7.6923076923076925 %


# Мужчины из 3 класса по возростным группам

In [23]:
male_all_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male")].shape[0] # Колличество мужчие из первого класса
male_all_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male")].shape[0]

male__12_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
male__12_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

male_12_20_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
male_12_20_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

male_20_30_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
male_20_30_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

male_30_40_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
male_30_40_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

male_40_50_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
male_40_50_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

male_50__pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
male_50__pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "male") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", male_all_pclass_3, male_all_pclass_3_surv, male_all_pclass_3_surv/male_all_pclass_3 * 100, "%", '\n',
      
      "до 12 лет", male__12_pclass_3, male__12_pclass_3_surv, male__12_pclass_3_surv/male__12_pclass_3 * 100, "%", '\n',

      "20 - 30 лет", male_20_30_pclass_3, male_20_30_pclass_3_surv, male_20_30_pclass_3_surv/male_20_30_pclass_3 * 100, "%", '\n',

      "30 - 40 лет", male_30_40_pclass_3, male_30_40_pclass_3_surv, male_30_40_pclass_3_surv/male_30_40_pclass_3 * 100, "%", '\n',

      "40 - 50 лет", male_40_50_pclass_3, male_40_50_pclass_3_surv, male_40_50_pclass_3_surv/male_40_50_pclass_3 * 100, "%", '\n',
      
      "50 лет и старше", male_50__pclass_3, male_50__pclass_3_surv, male_50__pclass_3_surv/male_50__pclass_3 *100, "%")

Всего 347 47 13.544668587896252 % 
 до 12 лет 25 9 36.0 % 
 20 - 30 лет 97 14 14.432989690721648 % 
 30 - 40 лет 49 7 14.285714285714285 % 
 40 - 50 лет 22 2 9.090909090909092 % 
 50 лет и старше 10 0 0.0 %


# Женщины из первого класса по возростным группам

In [24]:
female_all_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female")].shape[0] # Колличество мужчие из первого класса
female_all_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female")].shape[0]

female__12_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
female__12_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

female_12_20_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
female_12_20_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

female_20_30_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
female_20_30_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

female_30_40_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
female_30_40_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

female_40_50_pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
female_40_50_pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

female_50__pclass_1 = df[(df['Pclass'] == 1) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
female_50__pclass_1_surv = df[(df['Survived'] != False) & (df['Pclass'] == 1) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", female_all_pclass_1, female_all_pclass_1_surv, female_all_pclass_1_surv/female_all_pclass_1 * 100, "%", '\n',
      "до 12 лет", female__12_pclass_1, female__12_pclass_1_surv, female__12_pclass_1_surv/female__12_pclass_1 * 100, "%",'\n',
      "12 - 20 лет", female_12_20_pclass_1, female_12_20_pclass_1_surv, female_12_20_pclass_1_surv/female_12_20_pclass_1 * 100, "%", '\n',
      "20 - 30 лет", female_20_30_pclass_1, female_20_30_pclass_1_surv, female_20_30_pclass_1_surv/female_20_30_pclass_1 * 100, "%", '\n',
      "30 - 40 лет", female_30_40_pclass_1, female_30_40_pclass_1_surv, female_30_40_pclass_1_surv/female_30_40_pclass_1 * 100, "%", '\n',
      "40 - 50 лет", female_40_50_pclass_1, female_40_50_pclass_1_surv, female_40_50_pclass_1_surv/female_40_50_pclass_1 * 100, "%", '\n',
      "50 лет и старше", female_50__pclass_1, female_50__pclass_1_surv, female_50__pclass_1_surv/female_50__pclass_1 *100, "%")

Всего 94 91 96.80851063829788 % 
 до 12 лет 1 0 0.0 % 
 12 - 20 лет 13 13 100.0 % 
 20 - 30 лет 21 20 95.23809523809523 % 
 30 - 40 лет 24 24 100.0 % 
 40 - 50 лет 13 12 92.3076923076923 % 
 50 лет и старше 15 14 93.33333333333333 %


# Женщины из 2 класса по возростным группам

In [25]:
female_all_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female")].shape[0] # Колличество мужчие из первого класса
female_all_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female")].shape[0]

female__12_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
female__12_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

female_12_20_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
female_12_20_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

female_20_30_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
female_20_30_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

female_30_40_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
female_30_40_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

female_40_50_pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
female_40_50_pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

female_50__pclass_2 = df[(df['Pclass'] == 2) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
female_50__pclass_2_surv = df[(df['Survived'] != False) & (df['Pclass'] == 2) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", female_all_pclass_2, female_all_pclass_2_surv, female_all_pclass_2_surv/female_all_pclass_2 * 100, "%", '\n',
      
      "до 12 лет", female__12_pclass_2, female__12_pclass_2_surv, female__12_pclass_2_surv/female__12_pclass_2 * 100, "%", '\n',

      "20 - 30 лет", female_20_30_pclass_2, female_20_30_pclass_2_surv, female_20_30_pclass_2_surv/female_20_30_pclass_2 * 100, "%", '\n',

      "30 - 40 лет", female_30_40_pclass_2, female_30_40_pclass_2_surv, female_30_40_pclass_2_surv/female_30_40_pclass_2 * 100, "%", '\n',

      "40 - 50 лет", female_40_50_pclass_2, female_40_50_pclass_2_surv, female_40_50_pclass_2_surv/female_40_50_pclass_2 * 100, "%", '\n',
      
      "50 лет и старше", female_50__pclass_2, female_50__pclass_2_surv, female_50__pclass_2_surv/female_50__pclass_2 *100, "%")

Всего 76 70 92.10526315789474 % 
 до 12 лет 8 8 100.0 % 
 20 - 30 лет 28 25 89.28571428571429 % 
 30 - 40 лет 17 16 94.11764705882352 % 
 40 - 50 лет 10 9 90.0 % 
 50 лет и старше 6 5 83.33333333333334 %


# Женщины из 3 класса по возростным группам

In [26]:
female_all_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female")].shape[0] # Колличество мужчие из первого класса
female_all_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female")].shape[0]

female__12_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество мужчие до 12 лет из первого класса
female__12_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & (df['Age'] <= 12)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

female_12_20_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]
female_12_20_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 12) & (df['Age'] <= 20))].shape[0]

female_20_30_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]
female_20_30_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 20) & (df['Age'] <= 30))].shape[0]

female_30_40_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]
female_30_40_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 30) & (df['Age'] <= 40))].shape[0]

female_40_50_pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]
female_40_50_pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & ((df['Age'] > 40) & (df['Age'] <= 50))].shape[0]

female_50__pclass_3 = df[(df['Pclass'] == 3) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество мужчие до 12 лет из первого класса
female_50__pclass_3_surv = df[(df['Survived'] != False) & (df['Pclass'] == 3) & (df['Sex'] == "female") & (df['Age'] >= 50)].shape[0] # Колличество выжевших мужчие до 12 лет из первого класса

print("Всего", female_all_pclass_3, female_all_pclass_3_surv, female_all_pclass_3_surv/female_all_pclass_3 * 100, "%", '\n',
      
      "до 12 лет", female__12_pclass_3, female__12_pclass_3_surv, female__12_pclass_3_surv/female__12_pclass_3 * 100, "%", '\n',

      "20 - 30 лет", female_20_30_pclass_3, female_20_30_pclass_3_surv, female_20_30_pclass_3_surv/female_20_30_pclass_3 * 100, "%", '\n',

      "30 - 40 лет", female_30_40_pclass_3, female_30_40_pclass_3_surv, female_30_40_pclass_3_surv/female_30_40_pclass_3 * 100, "%", '\n',

      "40 - 50 лет", female_40_50_pclass_3, female_40_50_pclass_3_surv, female_40_50_pclass_3_surv/female_40_50_pclass_3 * 100, "%", '\n',
      
      "50 лет и старше", female_50__pclass_3, female_50__pclass_3_surv, female_50__pclass_3_surv/female_50__pclass_3 *100, "%")

Всего 144 72 50.0 % 
 до 12 лет 23 11 47.82608695652174 % 
 20 - 30 лет 32 16 50.0 % 
 30 - 40 лет 14 6 42.857142857142854 % 
 40 - 50 лет 8 0 0.0 % 
 50 лет и старше 1 1 100.0 %


# Обобщение
* 90-100% выжило женщин из первого и второго класса в любых возрастных группах
* 50% выжело женщин из 3 класса всех возростных групп 
## Мужчины из первого класса выжело 36.885% 
* до 12 лет 100.0 % 
* 12 - 20 лет 25.0 % 
* 20 - 30 лет 47.368 % 
* 30 - 40 лет 52.0 % 
* 40 - 50 лет 37.5 % 
* 50 лет и старше 20.69%
## Мужчины из 2 класса выжело 92.11% 
* до 12 лет 100.0 % 
* 20 - 30 лет 89.3 % 
* 30 - 40 лет 94.12 % 
* 40 - 50 лет 90.0 % 
* 50 лет и старше 83.33 %
## Мужчины из 3 класса выжело 13.54 % 
* до 12 лет 36.0 % 
* 20 - 30 лет 14.43 % 
* 30 - 40 лет 14.29 % 
* 40 - 50 лет 9.09 % 
* 50 лет и старше 0.0 %

# Вывод
## 1 место по выживаймости:
Выживаймость женщин и дете до 12 лет из 1 и 2 класса выше всего. Связано это с более благоприятными условиями чем в 3 классе и воспитанием. Женщин и детей спасают в первую очередь.
## 2 место по выживаймости:
Это мужчины из 2 класса так же из всех возростных групп. Думаю это связано с более высоким уровнем здоровья, лучше физическое и эмоциональное состояние в стрессовой ситуации. Все это в сравнении с мужчинами из 1 класса
## 3 место по выживаймости:
Женщины из 3 класса. Культура спосения женщин и детей так же присутствует, но доступность путей спасения ниже.
## 3 место по выживаймости:
Мужчины из 3 класса. Детей даже до 12лет этой категории спаслось меньше всего. Доступнасть спасения низкая, уровень культуры, паника.
## Интересное моменты
Судя по уровню выживаймости и если допустить связь с уровнем здоровья можно предположить что продуктивная жизнь мужчин из первого класса наченается в возросте 20 лет более позднее взрасление в сравнении с пассажирами из 2 класса и до 40 лет дальше происходит спад.
Мужчины из 2 класса демонстрируют самый продуктивный уровень жизни во всех возростах. Связано это с высокой физической активностью и достаточным уровнем жизни без свободного доступа к излишиствам.

Вот такие мысли.